utils/benchmark.py

```diff
@@ -116,6 +116,16 @@ def abort_maybe(self):
     def run(self, task_size: int, *args, **kwargs):
         raise NotImplementedError
 
+    def set_task_size(self, new_task_size):
+        """
+        A function appending new_task_size to the self._workload_size if not set yet or replacing the last value with the new_task_size if the last value is None.
+        Useful for models where the task_size is unknown before finishing the run (eg. text generation models where the number of tokens generated has an upper bound but can be lower)
+        """
+        if len(self._finish_times) - len(self._workload_size) == 1:
```

Comment:
Currently vulnerable to the wrong order of execution. Why not allow task size to be set before the given execution?
I.e.:

> assert abs(len(self._finish_times) - len(self._workload_size)) == 1
> self._workload_size.append(new_task_size)